java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessConnection.java

```diff
@@ -247,10 +249,8 @@ public void setReadOnly(boolean readOnly) throws SQLException {
                 Constants.SQLExceptionMessages.METHOD_CALLED_ON_OPEN_TRANSACTION);
         }
 
-        if (readOnly) {
-            throw new SQLFeatureNotSupportedException(Constants.SQLExceptionMessages.READ_ONLY);
-        }
-        this.readOnly = false;
+        // TODO implement read only properly
+        this.readOnly = readOnly;
```

Comment:
After thinking about this some more, I'm not too crazy about the API giving the impression that it supports a feature, yet in reality disregarding it silently.  I've never used the readonly flag myself... but if someone did, this approach could cause a very dangerous surprise.

I would say either go ahead and implement readonly (e.g. check this flag from `executeUpdate` as @tirsen suggests), or else throw an unsupported operation (or whatever) exception so that the truth is apparent.

I hate that this lack of support causes problems for downstream code outside of @tirsen 's control... and if this were a private fork for my own particular needs, then I might use an approach like this to solve that problem.  But it's a bit of a dangerous workaround to ship out to the public world.